src/integrated/land.jl

```diff
@@ -36,6 +36,58 @@ struct LandModel{
     canopy::VM
     "The snow model to be used"
     snow::SnM
+    function LandModel{FT}(
+        canopy::VM,
+        snow::SnM,
+        soil::SM,
+        soilco2::MM,
+    ) where {
+        FT,
+        MM <: Soil.Biogeochemistry.SoilCO2Model{FT},
+        SM <: Soil.EnergyHydrology{FT},
+        VM <: Canopy.CanopyModel{FT},
+        SnM <: Snow.SnowModel{FT},
+    }
+        prognostic_land_components = (:canopy, :snow, :soil, :soilco2)
+        top_soil_bc = soil.boundary_conditions.top
+        canopy_bc = canopy.boundary_conditions
+        snow_bc = snow.boundary_conditions
+
+        # Integrated model checks
+        @assert top_soil_bc.prognostic_land_components ==
+                prognostic_land_components
+        @assert canopy_bc.prognostic_land_components ==
+                prognostic_land_components
+        @assert snow_bc.prognostic_land_components == prognostic_land_components
+
+        @assert top_soil_bc.atmos == soilco2.drivers.atmos
+        @assert top_soil_bc.atmos == canopy_bc.atmos
+        @assert top_soil_bc.radiation == canopy_bc.radiation
+        @assert top_soil_bc.atmos == snow_bc.atmos
+        @assert top_soil_bc.radiation == snow_bc.radiation
+
+        @assert Domains.obtain_surface_domain(soil.domain) == canopy.domain
+        @assert Domains.obtain_surface_domain(soilco2.domain) == canopy.domain
+        @assert snow.domain == canopy.domain
+
+        @assert soil.parameters.earth_param_set ==
+                soilco2.parameters.earth_param_set
+        @assert soil.parameters.earth_param_set ==
+                canopy.parameters.earth_param_set
+        @assert soil.parameters.earth_param_set ==
+                snow.parameters.earth_param_set
+
+        # LandModel-specific checks
+        # Runoff and sublimation are also automatically included in the soil model
+        @assert RootExtraction{FT}() in soil.sources
+        @assert Soil.PhaseChange{FT}() in soil.sources
+        @assert canopy.hydraulics.transpiration isa
+                Canopy.PlantHydraulics.DiagnosticTranspiration{FT}
+        @assert canopy_bc.ground isa PrognosticSoilConditions{FT}
```

Comment:
I think this one needs to be PrognosticGroundConditions (soil+snow)

Reply:
I see, that wasn't clear to me. Maybe PrognosticSoilConditiions and PrognosticSoilSnowConditiions would be more clear

Reply:
Actually, do we even need this distinction? It looks like the only place we have different behavior for the two is in the ground albedo functions, and we also pass in `prognostic_land_components` - [here](https://github.com/CliMA/ClimaLand.jl/blob/f10d8f417484287ad2a5c425c1b74cbd7aa641ea/src/integrated/land_radiation.jl#L105-L141)

Reply:
oh! That would be a great change

Reply:
That sounds good! Here's an issue for it - #1272